### PR TITLE
aml: Implement DefOr, DefSubtract and DefLNot opcodes

### PR DESCRIPTION
**Pull request by alnyan**
This PR implements some simple opcodes currently missing, but required for real-hardware function.

### CODE REVIEW

File: aml/src/expression.rs [hunk 1]
```diff
@@ -91,6 +94,32 @@ where
...
+    'c: 'a,
+{
+    /*
+     * DefSubtract := 0x74 Operand Operand Target
+     * Operand := TermArg => Integer
+     */
+    opcode(opcode::DEF_SUBTRACT_OP)
+        .then(comment_scope(
+            DebugVerbosity::AllScopes,
+            "DefAdd",
```
Comment (IsaacWoods) [inline aml/src/expression.rs:108]: This should be `DefSubtract`

File: aml/src/expression.rs [hunk 2]
```diff
@@ -350,6 +405,26 @@ where
...
+     * DefLNot := 0x92 Operand
+     * Operand := TermArg => Integer
+     */
+    opcode(opcode::DEF_L_NOT_OP)
+        .then(comment_scope(
+            DebugVerbosity::AllScopes,
+            "DefLNot",
+            term_arg().map_with_context(|arg, context| {
+                let operand = try_with_context!(context, arg.as_bool());
+                (Ok(AmlValue::Boolean(operand)), context)
```
Comment (IsaacWoods) [inline aml/src/expression.rs:422]: I might just be missing it, but I don't see where the actual not is happening. Should this be `!operand`?

**Review by IsaacWoods — CHANGES_REQUESTED** [review]
Thanks for submitting this! Always good to get these sorts of opcodes out the way.

Just a few things:

**Comment by alnyan** [comment]
Yeah, sorry, these were pretty dumb mistakes I made while copy-pasting code :sweat_smile:

**Comment by IsaacWoods** [comment]
No worries; I've made plenty! Thanks!

**Review by IsaacWoods — APPROVED** [review]

**Pull request merged**